src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/Text/TextInterface/FontCollectionLoader.cs

```diff
@@ -0,0 +1,82 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using MS.Internal.Text.TextInterface.Interfaces;
+
+namespace MS.Internal.Text.TextInterface
+{
+    [ClassInterface(ClassInterfaceType.None)]
+    [ComVisible(true)]
+    internal unsafe class FontCollectionLoader : IDWriteFontCollectionLoaderMirror
+    {
+        private const int S_OK = unchecked((int)0L);
+        private const int E_INVALIDARG = unchecked((int)0x80070057L);
+
+        private readonly IFontSourceCollectionFactory _fontSourceCollectionFactory;
+        private readonly FontFileLoader _fontFileLoader;
+
+        public FontCollectionLoader()
+        {
+            Debug.Fail("Assertion failed");
```

Comment:
I'm not sure what's the reasoning behind this, the original code was like this so I added it. Maybe C++/CLI doesn't/didn't support private constructor or maybe it's because it's a COM object and it was to ensure that it wasn't created through COM using the default constructor ?